src/scenario.jl

```diff
@@ -580,40 +573,48 @@ function run_model(domain::Domain, param_set::NamedDimsArray, corals::DataFrame,
     # Pre-calculate proportion of survivers from wave stress
     # Sw_t = wave_damage!(cache.wave_damage, wave_scen, corals.wavemort90, n_species)
 
-    site_k_prop = max_cover'
-    absolute_k_area = site_k_area(domain)'  # max possible coral area in m^2
+    p.r .= corals.growth_rate
+    p.mb .= corals.mb_rate
+    ode_u = zeros(n_species, n_locs)
     growth::ODEProblem = ODEProblem{true}(growthODE, ode_u, tspan, p)
-    tmp::Matrix{Float64} = zeros(size(Y_cover[1, :, :]))  # temporary array to hold intermediate covers
+    alg_hint = Symbol[:nonstiff]
 
     area_weighted_TP = TP_data .* site_k_area(domain)
     TP_cache = similar(area_weighted_TP)
 
     # basal_area_per_settler is the area in m^2 of a size class one coral
     basal_area_per_settler = colony_mean_area(corals.mean_colony_diameter_m[corals.class_id.==1])
     for tstep::Int64 in 2:tf
-        p_step::Int64 = tstep - 1
-        Y_pstep .= Y_cover[p_step, :, :]
+        # Copy cover for previous timestep as basis for current timestep
+        C_t .= C_cover[tstep - 1, :, :]
 
-        # Calculates scope for coral fedundity for each size class and at each site.
-        fecundity_scope!(fec_scope, fec_all, fec_params_per_m², Y_pstep, total_loc_area)
+        # Calculates scope for coral fedundity for each size class and at each location
+        fecundity_scope!(fec_scope, fec_all, fec_params_per_m², C_t, loc_k_area)
 
-        site_coral_cover = sum(Y_pstep, dims=1)  # dims: nsites * 1
-        leftover_space_prop = relative_leftover_space(site_k_prop, site_coral_cover)
-        leftover_space_m² = leftover_space_prop .* total_loc_area
+        loc_coral_cover = sum(C_t, dims=1)  # dims: 1 * nsites
+        leftover_space_m² = relative_leftover_space(loc_coral_cover) .* loc_k_area
 
         # Recruitment represents additional cover, relative to total site area
-        # Gets used in ODE
-        p.rec .= settler_cover(fec_scope, TP_data, leftover_space_prop,
-            sim_params.max_settler_density, sim_params.max_larval_density, basal_area_per_settler)
+        # Recruitment/settlement occurs after the full moon in October/November
+        p.rec[:, valid_locs] .= settler_cover(
```

Comment:
Wasn't p.rec taken out of the ODE?

Reply:
Yes, it was - `p.rec` is now just re-used as a cache to store the updated values (down below it is used to update coral cover for time $t$).

Agree that it makes sense to move `rec` into the `cache` object, I guess I was being lazy.

Good pick up.